CONTRIBUTING.md

```diff
@@ -47,47 +47,48 @@ First, ensure that [Yarn](https://yarnpkg.com/) and
 
 ## The demo project
 
-The [demo project](demo) was initially forked from
-[dvc-checkpoints-mnist](https://github.com/iterative/dvc-checkpoints-mnist/tree/make_checkpoint),
-and is provided as a lightweight, convenient testbed to try the extension out
-with. It is not an exhaustive showcase of DVC's features, so testers are
-encouraged to try other DVC repositories -- especially real-world cases!
+The [demo project](demo) was initially forked from [dvc-checkpoints-mnist], and
+is provided as a lightweight, convenient testbed to try the extension out with.
+It is not an exhaustive showcase of DVC's features, so testers are encouraged to
+try other DVC repositories -- especially real-world cases!
 
-- Go to the `demo/` directory of this repo.
+- Install [Yarn](https://yarnpkg.com/) if needed. Go to the `demo/` directory of
+  this repo (`cd demo`) and run `yarn install` to set up the project.
 
-- It is recommended that you create the project's virtual environment with
-  Python 3.8 as some dependencies are not yet available for Python 3.9
+- Go back to the root of this repo (`cd ..`) and run `yarn setup:venv` to set up
+  a Python virtual environment in the demo project (in `demo/.env`).
 
-- `yarn install` is a prerequisite for automatically setting up the demo
-  project.
+  > Go to `demo/` if you want to use DVC from command line (terminal).
 
-- Run `yarn setup:venv` from the monorepo root (not `./demo`) to automatically
-  setup the Python virtual environment in the demo project.
+- Pull the project data with the extension (**DVC panel** in the _Source Code_
+  view) or with `.env/bin/dvc pull` from command line.
 
-- Once the environment has been setup you should navigate into the demo folder
-  activate the virtual environment and run `dvc pull`.
+- In order to run experiments in the demo project (from the **DVC View**), the
+  virtual env should be loaded through the Microsoft [Python extension] (please
+  install it), having this in `.vscode/settings.json`:
 
-- In order to run experiments in our demo project we require the Python virtual
-  environment to be loaded.
+  ```json
+  "python.defaultInterpreterPath": "demo/.env/bin/dvc"
+  ```
 
-- This will happen automatically IF the `ms-python.python` extension is
-  installed within the VS Code instance that you are developing against.
+[dvc-checkpoints-mnist]:
+  https://github.com/iterative/dvc-checkpoints-mnist/tree/make_checkpoint
+[python extension]:
+  https://marketplace.visualstudio.com/items?itemName=ms-python.python
 
 <a id='warning'></a>
 
-> **Warning**  
-> When using any project that relies on an isolated Python environment (`conda`,
-> `venv`, etc.), Microsoft's
-> [Python extension](https://github.com/Microsoft/vscode-python) is required.
+> **Warning** : When using any project that relies on an isolated Python
+> environment (`conda`, `venv`, etc.), Microsoft's Python extension is required.
 > It's used by this extension to locate and utilize the required environment.
```

Comment:
Why is this the case? Why should some other extension ever be required for this extension to work?

Reply:
this is fine, if you use Python it'll be installed. We need it since we are using its Python-specific APIs - like detecting and activating all these virtual envs